### PR TITLE
ODS-5554 Packages- EdFi.Ods.Extensions.Sk workflow added

### PR DESCRIPTION
**Pull request by semalaiappan**
image

### CODE REVIEW

File: sample-extension/EdFi.Ods.Extensions.Sk/EdFi.Ods.Extensions.Sk.nuspec [hunk 1]
```diff
@@ -1,12 +1,13 @@
...
 <package xmlns="http://schemas.microsoft.com/packaging/2010/07/nuspec.xsd">
   <metadata>
-    <id>EdFi.Ods.Extensions.Sk</id>
+    <id>EdFi.Suite3.Ods.Extensions.Sk</id>
     <version>0.0.0</version>
     <authors>Ed-Fi Alliance</authors>
     <owners>Ed-Fi Alliance</owners>
     <projectUrl>https://github.com/Ed-Fi-Alliance-OSS/Ed-Fi-Extensions</projectUrl>
     <requireLicenseAcceptance>false</requireLicenseAcceptance>
+	  <copyright>Copyright © 2021 Ed-Fi Alliance, LLC and Contributors</copyright>
```
Comment (bptillman) [inline sample-extension/EdFi.Ods.Extensions.Sk/EdFi.Ods.Extensions.Sk.nuspec]: This appears to have been indented further than the rest of the lines? Possibly due to a difference between tabs/spaces? Can you please fix it?

File: sample-extension/EdFi.Ods.Extensions.Sk/EdFi.Ods.Extensions.Sk.nuspec [hunk 2]
```diff
@@ -16,6 +17,6 @@
     <file src="Artifacts/**/*.sql" target="."/>
     <file src="Artifacts/**/*.json" target="."/>
     <file src="assemblyMetadata.json" target="."/>
-    <file src="bin/$configuration$/net6.0/EdFi.Ods.Extensions.Sk.dll" target="."/>
+    <file src="bin/$Configuration$/net6.0/EdFi.Ods.Extensions.Sk.dll" target="."/>
```
Comment (bptillman) [inline sample-extension/EdFi.Ods.Extensions.Sk/EdFi.Ods.Extensions.Sk.nuspec]: Is this change needed? I would have thought the previous way it was specified with a lowercase c would also work.

File: .github/workflows/Packages - EdFi.Ods.Extensions.SK.yml [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+
+name: Packages - EdFi.Ods.Extensions.Sk
+
+on:
+  pull_request:
+    branches: [main]
+  push:
+    branches:
+      - main
+  workflow_dispatch:
```
Comment (bptillman) [inline .github/workflows/Packages - EdFi.Ods.Extensions.SK.yml:14]: To reduce clean up work later as part of ODS-5589, please remove the inputs here since they are not needed.

File: .github/workflows/Packages - EdFi.Ods.Extensions.SK.yml [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+      shell: pwsh
+      run: |
+          $patternName = 'refs/heads/' +  '${{ env.CURRENT_BRANCH }}'
+          $is_pull_request_branch = 'False'
+          $is_pull_request_branch = git ls-remote --heads origin ${{ env.CURRENT_BRANCH }} | Select-String -Pattern $patternName -SimpleMatch -Quiet
+          if ($is_pull_request_branch -eq $true) {
+            git fetch origin ${{ env.CURRENT_BRANCH }}
+            git checkout ${{ env.CURRENT_BRANCH }}
+          }
+    - name: Register Ed-Fi Azure Artifacts Package Source
```
Comment (bptillman) [inline .github/workflows/Packages - EdFi.Ods.Extensions.SK.yml:74]: Good catch on finding how we needed to add this so that it could restore packages from the Azure Artifacts feed since it is not specified in the nuget config by default for this build

File: build.githubactions.ps1 [hunk 5]
```diff
@@ -0,0 +1,212 @@
...
+
+function Pack {
+    if ([string]::IsNullOrWhiteSpace($PackageName) -and [string]::IsNullOrWhiteSpace($NuspecFilePath)){
+        Invoke-Execute {
+            dotnet pack $ProjectFile -c $Configuration --output $packageOutput --no-build --verbosity normal -p:VersionPrefix=$version -p:NoWarn=NU5123
+        }
+    }
+    if ($NuspecFilePath -Like "*.nuspec" -and $PackageName -ne $null){
+       $nugetRoot = Split-Path -Path $NuspecFilePath -Parent
+       Set-Location -Path $nugetRoot  -PassThru
```
Comment (bptillman) [inline build.githubactions.ps1]: Do we need to keep the `-PassThru` that was added here? It will output the path, but Github logs look to truncate it so its not very useful for any debugging purposes, as you can see here:
image

**Review by bptillman — CHANGES_REQUESTED** [review]
The package contents look good, and there's only a few minor changes to make here and then we can consider this one done.

File: .github/workflows/Packages - EdFi.Ods.Extensions.SK.yml (hunk 4 above)
Comment (semalaiappan) [reply .github/workflows/Packages - EdFi.Ods.Extensions.SK.yml:74]: Thanks

File: sample-extension/EdFi.Ods.Extensions.Sk/EdFi.Ods.Extensions.Sk.nuspec (hunk 2 above)
Comment (semalaiappan) [reply sample-extension/EdFi.Ods.Extensions.Sk/EdFi.Ods.Extensions.Sk.nuspec]: yes I worked for lowercase

File: build.githubactions.ps1 (hunk 5 above)
Comment (semalaiappan) [reply build.githubactions.ps1]: I removed it . Thanks

File: .github/workflows/Packages - EdFi.Ods.Extensions.SK.yml (hunk 3 above)
Comment (semalaiappan) [reply .github/workflows/Packages - EdFi.Ods.Extensions.SK.yml:14]: I removed it . Thanks

**Comment by semalaiappan** [comment]
CodeQL Security Scan Pull request  build is failing from main https://github.com/Ed-Fi-Alliance-OSS/Starter-Kit-SEA-Modernization/actions/runs/3085378784/jobs/5205142728   will be handled in separate ticket

File: sample-extension/EdFi.Ods.Extensions.Sk/EdFi.Ods.Extensions.Sk.nuspec (hunk 1 above)
Comment (semalaiappan) [reply sample-extension/EdFi.Ods.Extensions.Sk/EdFi.Ods.Extensions.Sk.nuspec]: Fixed it .Thanks

**Review by bptillman — APPROVED** [review]
